common/src/main/java/com/microsoft/identity/common/internal/cache/SharedPreferencesAuthorityValidationMetadataCache.java

```diff
@@ -0,0 +1,94 @@
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.common.internal.cache;
+
+import android.content.Context;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.annotation.VisibleForTesting;
+import com.microsoft.identity.common.adal.internal.cache.IStorageHelper;
+import com.microsoft.identity.common.java.exception.ClientException;
+import static com.microsoft.identity.common.java.exception.ErrorStrings.ENVIRONMENT_CANNOT_BE_NULL_AS_A_AUTHORITY_VALIDATION_METADATA_KEY;
+import static com.microsoft.identity.common.java.exception.ErrorStrings.VALUE_CANNOT_BE_NULL_AS_A_AUTHORITY_VALIDATION_METADATA_VALUE;
+
+public class SharedPreferencesAuthorityValidationMetadataCache implements IAuthorityValidationMetadataCache {
+
+    private static final String CACHE_VALUE_SEPARATOR = "-";
+    private static final String AUTHORITY_VALIDATION_METADATA_CACHE_GUID = "33DD5583-1098-4617-AF07-2D327BC4C0E4";
+
+    /**
+     * The name of the SharedPreferences file on disk.
+     * This file is only used to store MSAL CPP authority validation metadata
+     */
+    private static final String DEFAULT_CPP_AUTHORITY_VALIDATION_METADATA_SHARED_PREFERENCES =
+            "com.microsoft.identity.client.cpp_authority_validation_metadata";
+
+    private final ISharedPreferencesFileManager mCppAuthorityValidationMetadataSharedPreferencesFileManager;
+
+    public SharedPreferencesAuthorityValidationMetadataCache(final Context context) {
+        // We don't need to encrypt this cache file, since the cache file does not contain any
+        // secrets and other apps do not have access to this cache file.
+        final IStorageHelper storageHelper = null;
+        mCppAuthorityValidationMetadataSharedPreferencesFileManager =
+                SharedPreferencesFileManager.getSharedPreferences(
+                        context,
+                        DEFAULT_CPP_AUTHORITY_VALIDATION_METADATA_SHARED_PREFERENCES,
+                        storageHelper
+                );
+    }
+
+    /**
+     * Generate cache Key for authority validation metadata.
+     *
+     * @param environment Environment
+     * @return String
+     */
+    private String generateAuthorityValidationMetadataKey(@NonNull final String environment) throws ClientException {
+        if (null == environment) {
+            throw new ClientException(ENVIRONMENT_CANNOT_BE_NULL_AS_A_AUTHORITY_VALIDATION_METADATA_KEY);
+        }
+        return AUTHORITY_VALIDATION_METADATA_CACHE_GUID + CACHE_VALUE_SEPARATOR + environment;
+    }
+
+    @Override
+    public void saveAuthorityValidationMetadata(@NonNull final String environment, @NonNull final String cacheValue) throws ClientException {
+        if (null == cacheValue) {
+            throw new ClientException(VALUE_CANNOT_BE_NULL_AS_A_AUTHORITY_VALIDATION_METADATA_VALUE);

```

Comment:
Same feedback here -- since you've explicitly declared `@NonNull` on your method, I would say throwing a checked Exception is unnecessary and instead `throw IllegalArgumentException`